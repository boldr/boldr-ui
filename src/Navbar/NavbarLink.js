/* @flow */
import * as React from 'react';
import classNames from 'classnames';
import { getActiveModifiers, removeActiveModifiers, createWrappedComponent } from '../util/boldrui';
import { getDomSafeProps } from '../util/helpers';
import { StyleClasses } from '../theme/styleClasses';

export type Props = {
  tag?: string,
  isHoverable?: boolean,
  hasDropdown?: boolean,
  render?: Function,
  className?: string,
  href?: string,
};

const BASE_ELEMENT = StyleClasses.NAVBAR_LINK;

export function NavbarLink({ tag = 'a', render, ...props }: Props) {
  const className = classNames(
    BASE_ELEMENT,
    {
      ...getActiveModifiers(props),
    },
    props.className,
  );

  if (render) {
    return render({ ...props, className });
  }

  const HTMLProps = getDomSafeProps(props, removeActiveModifiers);

  return React.createElement(tag, { ...HTMLProps, className });
}

export default createWrappedComponent(NavbarLink);
