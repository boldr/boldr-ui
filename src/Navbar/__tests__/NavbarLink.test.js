import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { NavbarLink } from '../NavbarLink';

describe('NavbarLink', () => {
  it('should render a custom component with NavbarLink props', () => {
    const customComponent = props => (
      <div>
        My Button <a {...props} />
      </div>
    );

    const customComponentRendered = (
      <div>
        My Button <a href="#" className="boldr-navbar__link" />
      </div>
    );

    const component = shallow(<NavbarLink href="#" render={customComponent} />);
    expect(component.contains(customComponentRendered)).toBe(true);
  });

  it('should render an anchor with .boldr-navbar__link', () => {
    const component = shallow(<NavbarLink href="#">My NavbarLink</NavbarLink>);
    expect(
      component.contains(
        <a href="#" className="boldr-navbar__link">
          My NavbarLink
        </a>,
      ),
    ).toBe(true);
  });

  it('should render a button with .boldr-navbar__link', () => {
    const component = shallow(<NavbarLink tag="button">My NavbarLink</NavbarLink>);
    expect(component.contains(<button className="boldr-navbar__link">My NavbarLink</button>)).toBe(
      true,
    );
  });

  it('should render a a with .boldr-navbar__link and modifiers', () => {
    const component = shallow(<NavbarLink isActive />);
    expect(component.is('a')).toBe(true);
    expect(component.hasClass('boldr-navbar__link')).toBe(true);
    expect(component.hasClass('is-active')).toBe(true);
  });

  it('should render a a with .boldr-navbar__link, modifiers and custom classNames', () => {
    const component = shallow(<NavbarLink isActive className="custom" />);
    expect(component.is('a')).toBe(true);
    expect(component.hasClass('boldr-navbar__link')).toBe(true);
    expect(component.hasClass('is-active')).toBe(true);
    expect(component.hasClass('custom')).toBe(true);
  });

  it('should render a a with .boldr-navbar__link and custom classNames', () => {
    const component = shallow(<NavbarLink className="custom" />);
    expect(component.is('a')).toBe(true);
    expect(component.hasClass('boldr-navbar__link')).toBe(true);
    expect(component.hasClass('custom')).toBe(true);
  });
  it('should match the snapshot', () => {
    const wrapper = shallow(<NavbarLink isActive />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
