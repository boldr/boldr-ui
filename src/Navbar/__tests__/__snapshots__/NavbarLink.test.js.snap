// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarLink should match the snapshot 1`] = `
<a
  className="boldr-navbar__link is-active"
/>
`;
